src/python/pants/backend/python/subsystems/python_setup.py

```diff
@@ -127,6 +132,20 @@ def resolver_allow_prereleases(self):
   def resolver_blacklist(self):
     return self.get_options().resolver_blacklist
 
+  @property
+  def build(self):
+    return self.get_options().build
+
+  @property
+  def default_precedence(self):
+    precedence = Sorter.DEFAULT_PACKAGE_PRECEDENCE[:]
+    if self.build is True:
+      allow_builds(precedence)
+    elif self.build is False:
+      no_allow_builds(precedence)
```

Comment:
`allow_builds` and `no_allow_builds` appear undefined here - did you mean `self._XXX`? this should also be assigning to `precedence` afaict.

also, if we're mutating the default based on options here then I think it's mislabling to call this "default precedence" - maybe just `def resolver_precedence`?.